### PR TITLE
feat: userop cli command in plugingen

### PR DESCRIPTION
**Pull request by denniswon**
Note: This is an “experimental” PR for the sake of unblocking other eng teams for faster and easier testing/debugging iterations. Not planned per Wallet Service Accounts (client) team current or future sprints. Again, please note that this is an “experimental” branch. cc: @moldy530 

https://alchemyinsights.slack.com/archives/C04RHSDSP9U/p1713169071062569

# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on enhancing the user operation functionality in the plugingen package. 

### Detailed summary
- Added `UserOpConfigJson` and `UserOpConfig` types
- Updated `rpcUrl` in `init.ts`
- Added `configFiles` array in `generate/index.ts`
- Added functions in `utils.ts` for loading JSON, fuzzy matching, and config validation
- Updated `findConfig.ts` to accept custom filenames
- Added commands for building, signing, and sending user operations in `cli.ts`
- Added examples and schemas for user operations
- Added functions in `client.ts` for creating a smart account client
- Added schemas for user operation data in `schema.ts`

> The following files were skipped due to too many changes: `packages/plugingen/src/commands/userop/schema.ts`, `packages/plugingen/src/commands/userop/index.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/plugingen/userop.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    },
+    "public": {
+      "rpcUrl": "https://arb-sepolia.g.alchemy.com/v2/MlNZ2V-DTCdvd3jqSAfaf3yDMIBciA8F"
+    }
+  },
+
+  "entrypoint": "0.6.0",
+
+  "account": {
+    "owner": "cook skin carbon unfold bundle half garbage scheme utility zero start culture argue armed clean",
```
Comment (moldy530) [inline packages/plugingen/userop.json:16]: Probably shouldn't have leaked this...

**Review by moldy530 — COMMENTED** [review]
This doesn't seem an appropriate function of the plugin gen cli tool? What's the goal here?

File: packages/plugingen/src/commands/generate/index.ts [hunk 2]
```diff
@@ -26,6 +26,15 @@ import { PluginActionsGenPhase } from "./phases/plugin-actions/index.js";
...
 import type { Phase, PhaseInput } from "./types.js";
 
+// Do not reorder
+// In order of preference files are checked
+export const configFiles = [
+  "plugingen.config.ts",
+  "plugingen.config.js",
+  "plugingen.config.mjs",
+  "plugingen.config.mts",
+];
```
Comment (moldy530) [inline packages/plugingen/src/commands/generate/index.ts:36]: find config already has this: https://github.com/alchemyplatform/aa-sdk/blob/main/packages/plugingen/src/utils/findConfig.ts

File: packages/plugingen/userop.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
+  "chain": "arbitrum-sepolia",
+
+  "connection_config": {
+    "bundler": {
+      "rpcUrl": "https://arb-sepolia.g.alchemypreview.com/v2/fGnvtpbRdda2Tk4ewN5nWxoX0dscaV_4"
+    },
+    "public": {
+      "rpcUrl": "https://arb-sepolia.g.alchemy.com/v2/MlNZ2V-DTCdvd3jqSAfaf3yDMIBciA8F"
```
Comment (moldy530) [inline packages/plugingen/userop.json:9]: definitely shouldn't have leaked this....

Comment (denniswon) [reply packages/plugingen/userop.json]: ah removed. thought I added to gitignore.

File: packages/plugingen/userop.json (hunk 1 above)
Comment (denniswon) [reply packages/plugingen/userop.json:12]: sorry. removed now, but probably should use another mnemonic.

File: packages/plugingen/src/commands/generate/index.ts (hunk 2 above)
Comment (denniswon) [reply packages/plugingen/src/commands/generate/index.ts:36]: yeah I use the same function, but just made the input file names to search for to the function.

**Comment by denniswon** [comment]
Note: This is an “experimental” PR for the sake of unblocking other eng teams for faster and easier testing/debugging iterations. Not planned per Wallet Service Accounts (client) team current or future sprints. Again, please note that this is an “experimental” branch. cc: @moldy530

**Pull request closed without merging**